### DISCUSSION TITLE
Where's the code that allows Lexical to handle long presses for display characters like “à” or "æ" ?

### DISCUSSION THREAD

**Discussion by AxelBriche** [facebook/lexical#6828] [Q&A]
![CleanShot 2024-11-13 at 22 46 50](https://github.com/user-attachments/assets/123c9227-67b6-44df-9412-35940c2c3c1e)

I press the “a” key, “a” is inserted, I click on “â” in the UI, “a” is replaced by “â”.
Where is the code that allows Lexical to handle this, please?

**Comment by etrepum** [comment]
There isn't any, this is all handled natively by the platform

**Reply by AxelBriche** [reply]
So where's the place where lexical don't block that action? Lexical usually does an event.preventDefault() when a key is pressed? Thank you very much.

**Comment by AxelBriche** [comment]
If not, please tell me where the entry point is when you press a key in the text editor. Thank you for your help.

**Reply by etrepum** [reply]
Lexical doesn't work like you think it does, there is no preventDefault happening here.

**Reply by etrepum** [reply]
The entry point is entirely within the browser and platform, the browser's support for input in contentEditable elements handles all of the hard stuff. Lexical is called later, from some other event, probably a mutation observer in this case.
